Assets/LeapMotionModules/UI/Scripts/ColliderUtil.cs

```diff
@@ -0,0 +1,281 @@
+    return (point - center).normalized * distance + center;
+  }
+
+
+  /* Given a collider and a position relative to the colliders transform, return whether or not the 
+   * position lies within the collider.  Can also provide the allowed distance tolerance
+   */
+  public static bool isInsideCollider(this Collider collider, Vector3 localPosition, float tolerance = 0.0f) {
+    SphereCollider sphereCollider = collider as SphereCollider;
+    if (sphereCollider != null) {
+      return sphereCollider.isPointInside(localPosition, tolerance);
+    }
+    BoxCollider boxCollider = collider as BoxCollider;
+    if (boxCollider != null) {
+      return boxCollider.isPointInside(localPosition, tolerance);
+    }
+    CapsuleCollider capsuleCollider = collider as CapsuleCollider;
+    if (capsuleCollider != null) {
+      return capsuleCollider.isPointInside(localPosition, tolerance);
+    }
+    Debug.LogWarning("ColliderUtil does not work for Mesh Colliders!");
+    return false;
+  }
+
+  /* Given a collider and a position reltative to the colliders transform, return the point on the
+   * surface of the collider closest to the position
+   */
+  public static Vector3 closestPointOnSurface(this Collider collider, Vector3 localPosition, float extrude = 0.0f) {
+    SphereCollider sphereCollider = collider as SphereCollider;
+    if (sphereCollider != null) {
+      return sphereCollider.closestPointOnSurface(localPosition, extrude);
+    }
+    BoxCollider boxCollider = collider as BoxCollider;
+    if (boxCollider != null) {
+      return boxCollider.closestPointOnSurface(localPosition, extrude);
+    }
+    CapsuleCollider capsuleCollider = collider as CapsuleCollider;
+    if (capsuleCollider != null) {
+      return capsuleCollider.closestPointOnSurface(localPosition, extrude);
+    }
+    Debug.LogWarning("ColliderUtil does not work for Mesh Colliders!");
+    return Vector3.zero;
+  }
+
+  /* Given a GameObject and a position in global space, return whether or not the position lies within
+   * one of the colliders on the GameObject or one of it's children.  Can also provide the allowed 
+   * distance tolerance
+   */
+  public static bool isInsideColliders(GameObject obj, Vector3 globalPosition, float tolerance = 0.0f) {
+    obj.GetComponentsInChildren<Collider>(false, _colliderList);
+    foreach (Collider collider in _colliderList) {
+      Vector3 localPosition = collider.transform.InverseTransformPoint(globalPosition);
+      if (collider.isInsideCollider(localPosition, tolerance)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /* Given a GameObject and a position in global space, return a point on the surface of one of the
+   * colliders on the GameObject or one of it's children that is closest to the given position
+   */
+  public static Vector3 closestPointOnSurfaces(GameObject obj, Vector3 globalPosition, float extrude = 0.0f) {
+    obj.GetComponentsInChildren<Collider>(false, _colliderList);
```

Comment:
Same idea here, list of colliders instead of a gameObject for efficiency (and greater flexibility!), this method will currently count trigger colliders!